services/testing-framework/claudie_test.go

```diff
@@ -126,147 +121,149 @@ func testClaudie(ctx context.Context) error {
-	if err != nil {
-		log.Fatal().Msgf("Failed to create client connection to context-box : %v", err)
-	}
-
-	// Creating the client
-	c := pb.NewContextBoxServiceClient(cc)
-	return c, cc
-}
-
-// applyTestSet function will apply test set sequentially to Claudie
-func applyTestSet(ctx context.Context, setName string, c pb.ContextBoxServiceClient) error {
+// processTestSet function will apply test set sequentially to Claudie
+func processTestSet(ctx context.Context, setName string, c pb.ContextBoxServiceClient, testFunc func(ctx context.Context, c *pb.Config) error) error {
+	// Set errCleanUp to clean up the infra on failure
+	var errCleanUp, errIgnore error
 	idInfo := idInfo{id: "", idType: -1}
-
 	pathToTestSet := filepath.Join(testDir, setName)
 	log.Info().Msgf("Working on the test set:%s", pathToTestSet)
 
-	manifestFiles, err := os.ReadDir(pathToTestSet)
-	if err != nil {
-		return fmt.Errorf("error while trying to read test manifests in %s : %w", pathToTestSet, err)
+	// Defer clean up function
+	defer func() {
+		if errCleanUp != nil {
+			if autoCleanUpFlag == "TRUE" {
+				log.Info().Msgf("Deleting infra even after error due to flag \"-auto-clean-up\" set to %v", autoCleanUpFlag)
+				// delete manifest from DB to clean up the infra
+				if err := cleanUp(setName, idInfo.id, c); err != nil {
+					log.Error().Msgf("error while cleaning up the infra for test set %s : %v", setName, err)
+				}
+			}
+		}
+	}()
+
+	manifestFiles, errIgnore := os.ReadDir(pathToTestSet)
+	if errIgnore != nil {
+		return fmt.Errorf("error while trying to read test manifests in %s : %w", pathToTestSet, errIgnore)
 	}
 
 	for _, manifest := range manifestFiles {
-		// https://github.com/Berops/claudie/pull/243#issuecomment-1218237412
-		if manifest.IsDir() || manifest.Name()[0:1] == "." {
-			continue
+		// Apply test set manifest
+		if errIgnore = applyManifest(setName, pathToTestSet, manifest, &idInfo, c); errIgnore != nil {
+			// https://github.com/berops/claudie/pull/243#issuecomment-1218237412
+			if errors.Is(errIgnore, errHidden) {
+				continue
+			}
+			return fmt.Errorf("error applying test set %s, manifest %s from %s : %w", manifest.Name(), setName, manifest.Name(), errIgnore)
 		}
 
-		// create a path and read the file
-		manifestPath := filepath.Join(pathToTestSet, manifest.Name())
-		yamlFile, err := os.ReadFile(manifestPath)
-		if err != nil {
-			return fmt.Errorf("error while reading the manifest %s : %w", manifestPath, err)
-		}
-		manifestName, err := getManifestName(yamlFile)
-		if err != nil {
-			return fmt.Errorf("error while getting the manifest name from %s : %w", manifestPath, err)
+		// Wait until test manifest has been processed
+		if errCleanUp = configChecker(ctx, c, pathToTestSet, manifest.Name(), idInfo); errCleanUp != nil {
+			if errors.Is(errCleanUp, errInterrupt) {
+				log.Warn().Msgf("Testing-framework received interrupt signal, aborting test checking")
+				// Do not return error, since it was an interrupt
```

Comment:
>we return nil here because we want to convey to further up the chain that this test has not failed but has instead been interrupted, such as due to a timeout?

Yes, if the test is interrupted, it does not mean it failed.

>what could be the/other reasons for an interrupt here?

The pod getting deleted due to various reasons, i.e. deleted job, rescaling of the cluster, manual deletion.... 

>would this somehow be used to signal our CI that the reason this failed was that it had been killed instead of a program error?

I am not sure how we would convey a signal from the testing framework pod to the CI.

Reply:
> Yes, if the test is interrupted, it does not mean it failed.
> > what could be the/other reasons for an interrupt here?
>
> The pod getting deleted due to various reasons, i.e. deleted job, rescaling of the cluster, manual deletion....

right, I see.

> I am not sure how we would convey a signal from the testing framework pod to the CI.

I mean, just by returning a status code from the testing framework, as I believe it currently works:
* 0 for success
* 1 for error

so for the interrupt it would either be error or yet something different.

my question would have been rather better phrased as "*is* this currently used to signal our CI...", to which I assume now the answer would be "no".